modules/pam_motd/pam_motd.c

```diff
@@ -282,6 +282,72 @@ static void try_to_display_directories_with_overrides(pam_handle_t *pamh,
     _pam_drop(dirscans);
 }
 
+static int drop_privileges(pam_handle_t *pamh, struct pam_modutil_privs *privs)
+{
+    struct passwd *pw;
+    const char *username;
+    int retval;
+
+    retval = pam_get_user(pamh, &username, "key user");
```

Comment:
Apparently, I missed this "key user" argument.
I suppose NULL should be passed to pam_get_user instead.

Reply:
Ah, good catch. Will you submit a patch?

Reply:
Thank you! I think I didn't use the correct example to drop/regain privileges.

Reply:
Created PR: https://github.com/linux-pam/linux-pam/pull/282